client/client.go

```diff
@@ -418,17 +419,59 @@ func (r *NotaryRepository) ListTargets() ([]*Target, error) {
 		return nil, err
 	}
 
+	if len(roles) == 0 {
+		roles = []string{data.CanonicalTargetsRole}
+	}
+	targets := make(map[string]*Target)
+	for _, role := range roles {
+		// we don't need to do anything special with removing role from
+		// roles because listSubtree always processes role and only excludes
+		// descendent delegations that appear in roles.
+		r.listSubtree(targets, role, roles...)
```

Comment:
That is correct. I also think that behaviour is correct. By passing that list of roles, you're explicitly saying you want `targets/a/b` to be treated as lower priority than `targets/x`, even though the rest of the `targets/a` subtree is treated as higher priority than `targets/x`